cross-chain/arbitrum/deploy_l2/02_deploy_arbitrum_wormhole_gateway.ts

```diff
@@ -0,0 +1,74 @@
+import { smock } from "@defi-wonderland/smock"
+
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+import type { IWormholeTokenBridge, IERC20Upgradeable } from "../typechain"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { ethers, getNamedAccounts, helpers, deployments } = hre
+  const { log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const L2TokenBridge = await deployments.getOrNull("TokenBridge") // L2
+  const WormholeTBTC = await deployments.getOrNull("WormholeTBTC") // L2
+  const ArbitrumTBTC = await deployments.get("ArbitrumTBTC")
+
+  let tokenBridgeAddress = ""
+  if (L2TokenBridge && helpers.address.isValid(L2TokenBridge.address)) {
+    log(`using existing L2 TokenBridge at ${L2TokenBridge.address}`)
+    tokenBridgeAddress = L2TokenBridge.address
+  } else {
+    // For local development
+    const FakeWormholeTokenBridge = await smock.fake<IWormholeTokenBridge>(
+      "IWormholeTokenBridge"
+    )
+    tokenBridgeAddress = FakeWormholeTokenBridge.address
+  }
+
+  let wormholeTBTCAddress = ""
+  if (WormholeTBTC && helpers.address.isValid(WormholeTBTC.address)) {
+    log(`using existing L2 WormholeTBTC at ${WormholeTBTC.address}`)
+    wormholeTBTCAddress = WormholeTBTC.address
+  } else {
+    // For local development
+    const TestERC20 = await smock.fake<IERC20Upgradeable>("IERC20Upgradeable")
+    wormholeTBTCAddress = TestERC20.address
+  }
```

Comment:
I am confused about this block. It looks like the only occurrence of `smock` is in this file.

Even if we wanted to use a mock for tests instead of a real implementation, I think it should happen outside of deployment scripts, in the specific unit test file.

Reply:
For `Goerli` and `Mainnet` we already have addresses of the `tokenBridge` and `wormholeTBTC`. They are in the `external` dir. However, when we run the deployment scripts or tests, hardhat will look at the `external` dir first and won't find any for `localhost/hardhat` network and complain about it. For the local run, it can be really any address. I decided to fake the contracts just in case we need to mock functions in the future for testing purposes. But we don't do mock anything now, so I guess we can simplify it to:
- not faking but deploy contracts locally for `tokenBridge` and `wormholeTBTC` and fetch their addresses
- in the `hardhat.config.ts` we can add `tokenBridge` and `wormholeTBTC` under the `namedAccounts` for the `hardhat` network. That's the most simple one.
```
    l2TokenBridge: {
      default: 3,
    },
    wormholeTBTC: {
      default: 4,
    }
```
Any preference?

Reply:
I would be fine with just faking their addresses. This is a simple module with no logic. All unit tests are only about the deployment.

Reply:
done in [0e5f877](https://github.com/keep-network/tbtc-v2/pull/554/commits/0e5f877ac5daf42ebf65eefa7d78282ff9b68e24)